tools/latency_logging/print_timestamps.py

```diff
@@ -0,0 +1,216 @@
+      frame_id = translate(float(jmsg["info"]["logMonoTime"]))
+      if frame_id == -1:
+        failed_transl += 1
+        continue
+      service_intervals[frame_id][SERVICES[-1]]["End"].append(float(jmsg["time"]))
+  return (service_intervals, frame_mismatches, internal_durations, failed_transl)
+
+def get_empty_data(service_intervals):
+  for frame_id in service_intervals.keys():
+    for service in SERVICES:
+      if service not in service_intervals[frame_id]:
+        yield frame_id
+
+def exclude_bad_data(frame_ids, service_intervals):
+  for frame_id in frame_ids:
+    del service_intervals[frame_id]
+
+def find_frame_id(time, service):
+  prev_service = SERVICES[SERVICES.index(service)-1]
+  for frame_id, intervals in service_intervals.items():
+    if service == 'camerad':
+      if min(intervals['camerad']["Start"]) <= time <= min(intervals[service]["End"]):
+        return frame_id
+    else:
+      if min(intervals[prev_service]["End"]) <= time <= max(intervals[service]["End"]):
+        return frame_id
+  return -1
+
+def fill_intervals(timestamp_logs, service_intervals):
+  t0 = min(service_intervals[min(service_intervals.keys())]['camerad']["Start"])
+  failed_inserts = 0
+  for jmsg in timestamp_logs:
+    service = jmsg['ctx']['daemon']
+    event = jmsg['msg']['timestamp']['event']
+    time = float(jmsg['msg']['timestamp']['time'])
+    if time < t0:
+      continue
+    frame_id = find_frame_id(time, service) 
+    if frame_id != -1:
+      service_intervals[frame_id][service][event].append(time)
+    else:
+      failed_inserts +=1
+  return failed_inserts
+
+def print_timestamps(service_intervals, relative_self):
+  t0 = min([min([min(times) for times in events.values()]) for events in service_intervals[min(service_intervals.keys())].values()])
+  for frame_id, services in service_intervals.items():
+    print('='*80)
+    print("Frame ID:", frame_id)
+
+    print("Timestamps:")
+    if relative_self:
+      t0 = min([min([min(times) for times in events.values()]) for events in services.values()])
+    for service in SERVICES:
+      events = service_intervals[frame_id][service]
+      print("  "+service)  
+      for event, times in sorted(events.items(), key=lambda x: x[1][-1]):
+        times = [(time-t0)/1e6 for time in times]
+        print("    "+'%-50s%-50s' %(event, str(times)))  
+
+    print("Internal durations:")
+    for service, events in dict(internal_durations)[frame_id].items():
+      print("  "+service)  
+      for event, times in dict(events).items():
+        print("    "+'%-50s%-50s' %(event, str(times)))  
+
+def graph_timestamps(service_intervals, relative_self):
+  t0 = min([min([min(times) for times in events.values()]) for events in service_intervals[min(service_intervals.keys())].values()])
+  gnt = plt.subplots()[1]
+  gnt.set_xlim(0, 150 if relative_self else 750)
+  gnt.set_ylim(0, len(service_intervals))
+  y = 0
+  for frame_id, services in service_intervals.items():
```

Comment:
use enumerate instead
```suggestion
  for y, frame_id, services in enumerate(service_intervals.items()):
```